src/core/src/op/group_normalization.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/group_normalization.hpp"
+
+#include "itt.hpp"
+#include "ngraph/validation_util.hpp"
+#include "openvino/core/attribute_visitor.hpp"
+
+using namespace ov;
+
+op::v12::GroupNormalization::GroupNormalization(const Output<Node>& data,
+                                                const Output<Node>& scale,
+                                                const Output<Node>& bias,
+                                                int64_t num_groups,
+                                                double epsilon)
+    : Op({data, scale, bias}),
+      m_num_groups{num_groups},
+      m_epsilon{epsilon} {
+    constructor_validate_and_infer_types();
+}
+
+bool op::v12::GroupNormalization::visit_attributes(AttributeVisitor& visitor) {
+    OV_OP_SCOPE(v12_GroupNormalization_visit_attributes);
+    visitor.on_attribute("num_groups", m_num_groups);
+    visitor.on_attribute("epsilon", m_epsilon);
+    return true;
+}
+
+void op::v12::GroupNormalization::validate_and_infer_types() {
+    OV_OP_SCOPE(v12_GroupNormalization_validate_and_infer_types);
+    const auto data_partial_shape = get_input_partial_shape(0);
+    const auto data_rank = data_partial_shape.rank();
+    const auto scale_partial_shape = get_input_partial_shape(1);
+    const auto bias_partial_shape = get_input_partial_shape(2);
+
+    NODE_VALIDATION_CHECK(this,
+                          scale_partial_shape.rank().compatible(Dimension{1}),
+                          "The scale input is required to be 1D");
+    NODE_VALIDATION_CHECK(this,
+                          bias_partial_shape.rank().compatible(Dimension{1}),
+                          "The bias input is required to be 1D");
+
+    if (data_rank.is_static() && data_rank.get_length() >= 2) {
+        const auto channels_dim = data_partial_shape[1];
```

Comment:
- The value of `num_groups` is not validated, but according to the spec it can't be lower than `1`.
- Also (not described by spec), is the result of static `channels_dim % num_groups` expected to be zero 
or rounded down like `channels_dim // num_groups`? 
- What if the `num_groups` is bigger than `channels_dim`?

Reply:
I've done some research and some experiments and it seems that we should assert that the channels dimension can be evenly divided by the number of groups. Additionally we should limit the upper bound value of `num_groups` to the value of `channels_dim`.

Reply:
Okay, it should be reflected in the ov GroupNormalization-12 specification.